### PR TITLE
(FACT-891) Extend check to unbreak compilation on OpenBSD

### PR DESCRIPTION
**Pull request by jasperla**
It'd be great if this could be committed while the underlying issues as described in FACT-891 are being resolved.

### CODE REVIEW

File: lib/src/util/posix/dynamic_library.cc [hunk 1]
```diff
@@ -41,10 +41,10 @@ namespace facter { namespace util {
         close();
 
         // Don't actually perform a load to determine if it is already loaded
-#ifdef _AIX
+#if defined(_AIX) || !defined(RTLD_NOLOAD)
```
Comment (peterhuene) [inline lib/src/util/posix/dynamic_library.cc]: Can probably remove the `_AIX` check now, as `RTLD_NOLOAD` won't be defined in either case.

I think we can probably remove the usage of `RTLD_NOLOAD` entirely, as we search for exported ruby symbols in POSIX and for loaded modules by name on Windows to find the existing library before attempting to load it.  This code existed before those checks were made.

**Comment by jasperla** [comment]
Ok, for the moment I've adjusted the checks. Though your rationale for removing `RTLD_NOLOAD` entirely makes sense.

**Comment by MikaelSmith** [comment]
:+1:

**Comment by MikaelSmith** [comment]
Actually, @branan should this work for AIX?

**Comment by branan** [comment]
This will also work on AIX, yes.

Obviously the issues with facter loading ruby (and vice-versa) won't be resolved, and this is going to behave badly in those situations.

**Pull request merged**